students/rgpag/lesson04/trigrams.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+import random
+
+# txt file to be used
+text_in = 'sherlock_full.txt'
+f = open(text_in)
+msg = f.read()
+f.close
+
+# manipulate text file to be more trigram friendly
+string = msg.lower()
+string = string.replace("\n", " ")
+string = string.replace("--", " ")
+string = string.replace("-", " ")
+spl_string = string.split(" ")
+
+# init vals
+str_dict = {}
+i = 0
+j = 0
+quit = 0
+
+# build trigram dictionary of keys
+for i in range(0, len(spl_string)-2):
+    key = spl_string[i] + " " + spl_string[i+1]
+    str_dict[key] = []
+
+# append values
+for i in range(0, len(spl_string)-2):
+    key = spl_string[i] + " " + spl_string[i+1]
+    str_dict[key].append(spl_string[i+2])
```

Comment:
Check out the `dict.setdefault()` method for a way to accomplish the same thing without looping through twice